tensorflow/tools/dist_test/python/mnist_replica.py

```diff
@@ -94,32 +93,48 @@ def main(unused_argv):
   if FLAGS.download_only:
     sys.exit(0)
 
+  if FLAGS.job_name is None or FLAGS.job_name == "":
+    raise ValueError("Must specify an explicit `job_name`")
+  if FLAGS.task_index is None or FLAGS.task_index =="":
+    raise ValueError("Must specify an explicit `task_index`")
+
   print("job name = %s" % FLAGS.job_name)
   print("task index = %d" % FLAGS.task_index)
   
   #Construct the cluster and start the server
   ps_spec = FLAGS.ps_hosts.split(",")
   worker_spec = FLAGS.worker_hosts.split(",")
 
+  # Get the number of workers 
+  num_workers = len(worker_spec)
+
```

Comment:
Get the `num_workers` from `worker_hosts` and delete the redundant flags `num_workers`